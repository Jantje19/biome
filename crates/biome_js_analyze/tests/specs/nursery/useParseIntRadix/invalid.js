var num = parseInt("071");

var num = parseInt(someValue);

var num = parseInt("071", "abc");

var num = parseInt("071", 37);

var num = parseInt();

var num = Number.parseInt("10");
