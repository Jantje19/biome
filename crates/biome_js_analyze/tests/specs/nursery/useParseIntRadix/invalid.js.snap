---
source: crates/biome_js_analyze/tests/spec_tests.rs
expression: invalid.js
snapshot_kind: text
---
# Input
```jsx
var num = parseInt("071");

var num = parseInt(someValue);

var num = parseInt("071", "abc");

var num = parseInt("071", "8");

var num = parseInt("071", 37);

var num = parseInt();

var num = Number.parseInt("10");

var num = parseInt("0xF", 16n);

var num = parseInt("071", "abc", 3);

```

# Diagnostics
```
invalid.js:1:11 lint/nursery/useParseIntRadix  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Missing radix parameter
  
  > 1 │ var num = parseInt("071");
      │           ^^^^^^^^^^^^^^^
    2 │ 
    3 │ var num = parseInt(someValue);
  
  i Add a non-fractional number between 2 and 36
  
  i Unsafe fix: Add a radix of 10
  
    1 │ var·num·=·parseInt("071",·10);
      │                         ++++  

```

```
invalid.js:3:11 lint/nursery/useParseIntRadix  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Missing radix parameter
  
    1 │ var num = parseInt("071");
    2 │ 
  > 3 │ var num = parseInt(someValue);
      │           ^^^^^^^^^^^^^^^^^^^
    4 │ 
    5 │ var num = parseInt("071", "abc");
  
  i Add a non-fractional number between 2 and 36
  
  i Unsafe fix: Add a radix of 10
  
    3 │ var·num·=·parseInt(someValue,·10);
      │                             ++++  

```

```
invalid.js:5:11 lint/nursery/useParseIntRadix ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Invalid radix parameter
  
    3 │ var num = parseInt(someValue);
    4 │ 
  > 5 │ var num = parseInt("071", "abc");
      │           ^^^^^^^^^^^^^^^^^^^^^^
    6 │ 
    7 │ var num = parseInt("071", "8");
  
  i Radix must be a non-fractional number between 2 and 36
  

```

```
invalid.js:7:11 lint/nursery/useParseIntRadix ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Invalid radix parameter
  
    5 │ var num = parseInt("071", "abc");
    6 │ 
  > 7 │ var num = parseInt("071", "8");
      │           ^^^^^^^^^^^^^^^^^^^^
    8 │ 
    9 │ var num = parseInt("071", 37);
  
  i Radix must be a non-fractional number between 2 and 36
  

```

```
invalid.js:9:11 lint/nursery/useParseIntRadix ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Invalid radix parameter
  
     7 │ var num = parseInt("071", "8");
     8 │ 
   > 9 │ var num = parseInt("071", 37);
       │           ^^^^^^^^^^^^^^^^^^^
    10 │ 
    11 │ var num = parseInt();
  
  i Radix must be a non-fractional number between 2 and 36
  

```

```
invalid.js:11:11 lint/nursery/useParseIntRadix ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Missing parameters
  
     9 │ var num = parseInt("071", 37);
    10 │ 
  > 11 │ var num = parseInt();
       │           ^^^^^^^^^^
    12 │ 
    13 │ var num = Number.parseInt("10");
  
  i Will always return NaN
  

```

```
invalid.js:13:11 lint/nursery/useParseIntRadix  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Missing radix parameter
  
    11 │ var num = parseInt();
    12 │ 
  > 13 │ var num = Number.parseInt("10");
       │           ^^^^^^^^^^^^^^^^^^^^^
    14 │ 
    15 │ var num = parseInt("0xF", 16n);
  
  i Add a non-fractional number between 2 and 36
  
  i Unsafe fix: Add a radix of 10
  
    13 │ var·num·=·Number.parseInt("10",·10);
       │                               ++++  

```

```
invalid.js:15:11 lint/nursery/useParseIntRadix ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Invalid radix parameter
  
    13 │ var num = Number.parseInt("10");
    14 │ 
  > 15 │ var num = parseInt("0xF", 16n);
       │           ^^^^^^^^^^^^^^^^^^^^
    16 │ 
    17 │ var num = parseInt("071", "abc", 3);
  
  i Radix must be a non-fractional number between 2 and 36
  

```

```
invalid.js:17:11 lint/nursery/useParseIntRadix ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Invalid radix parameter
  
    15 │ var num = parseInt("0xF", 16n);
    16 │ 
  > 17 │ var num = parseInt("071", "abc", 3);
       │           ^^^^^^^^^^^^^^^^^^^^^^^^^
    18 │ 
  
  i Radix must be a non-fractional number between 2 and 36
  

```
